### PR TITLE
using base32 encoding to avoid underscores and hyphens

### PR DESCRIPTION
**Pull request by 2is10**
@andrewconner

### CODE REVIEW

**Comment by andrewconner** [comment]
Looks good. Merge conflict (I made an AES instance in the config, tagged you on PR)

File: kifi-backend/modules/common/app/com/keepit/common/crypto/ModelWithPublicId.scala [hunk 1]
```diff
@@ -39,6 +38,8 @@ object PublicId {
 
     override def unbind(key: String, id: PublicId[T]): String = id.toString
   }
+
+  private[crypto] val coder = BaseEncoding.base32().lowerCase().omitPadding()
```
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/common/crypto/ModelWithPublicId.scala:42]: not sure whether we want lowercase or not. i think it helps with readability, but we may prefer having it look like a solid opaque block. i don’t feel strongly either way.

Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/crypto/ModelWithPublicId.scala:42]: lower is good with me

File: kifi-backend/modules/common/app/com/keepit/common/crypto/ModelWithPublicId.scala [hunk 2]
```diff
@@ -61,11 +62,11 @@ trait ModelWithPublicIdCompanion[T <: ModelWithPublicId[T]] {
...
   def decode(publicId: String)(implicit config: PublicIdConfiguration): Try[Id[T]] = {
-    val reg = raw"^$prefix(.*)$$".r
-    Try {
-      reg.findFirstMatchIn(publicId).map(_.group(1)).map { identifier =>
-        Id[T](config.aes64bit.decrypt(Base64.decodeBase64(identifier)))
-      }.get
+    if (publicId.startsWith(prefix)) {
+      Try(Id[T](config.aes64bit.decrypt(PublicId.coder.decode(publicId.substring(prefix.length)))))
+    } else {
+      Failure(new IllegalArgumentException(s"Expected $publicId to start with $prefix"))
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/crypto/ModelWithPublicId.scala:68]: Better!

**Pull request merged**